src/modules/systemlib/param/param.c

```diff
@@ -127,6 +127,7 @@ get_param_info_count(void)
 {
 	/* Singleton creation of and array of bits to track changed values */
 	if (!param_changed_storage) {
+		/* Note that we have a (highly unlikely) race condition here: in the worst case the allocation is done twice */
```

Comment:
I thought about this, but that's really a chicken-egg problem: adding a singleton requires synchronization itself to be thread-safe.
And static global object initializers should be avoided (Not sure but I thought they are not supported on NuttX)

I'll look into it but sched_lock is really only useful for NuttX.